packages/experimental/src/components/EmptyState/EmptyState.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright IBM Corp. 2020, 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'carbon-components-react';
+
+import { expPrefix } from '../../global/js/settings';
+
+export const EmptyState = ({
+  action,
+  heading,
+  subtext,
+  illustration,
+  illustrationSize,
+  illustrationTheme,
+}) => {
+  return (
+    <div className={`${expPrefix}-empty-state`}>
+      {illustration && typeof illustration === 'string' && (
+        <img
+          src={require(`./EmptyStateIllustrations/${illustrationTheme}/${illustration}.svg`)}
+          alt=""
+          className={[
+            `${expPrefix}-empty-state-illustration`,
+            `${expPrefix}-empty-state-illustration--${illustrationSize}`,
+          ].join(' ')}
+        />
+      )}
+      <h3 className={`${expPrefix}-header`}>{heading}</h3>
+      {typeof subtext !== 'string' ? (
+        subtext
+      ) : (
+        <p className={`${expPrefix}-subtext`}>{subtext}</p>
+      )}
+      {action && action.text && (
+        <Button
+          className={`${expPrefix}-empty-state-action-button`}
+          kind={action.type || 'tertiary'}
+          onClick={action.actionHandler}
+          renderIcon={action.renderIcon || null}>
+          {action.text}
+        </Button>
+      )}
+    </div>
+  );
+};
+
+EmptyState.propTypes = {
+  /**
+   * Empty state action button
+   */
+  action: PropTypes.shape({
```

Comment:
I'm curious about your choice to pass an object instead of passing the attributes separately. I guess it's more of a personal preference, but why do that instead of something like `actionText` `actionType` etc? I personally think it's more clear to the consumer and less prone to bugs to pass props individually instead of as an object. I don't know if anyone else wants to comment on this, but I think it would be interesting to discuss.

Reply:
Yeah, I agree that it would probably be clearer. At first I didn't know if there were going to be multiple actions but I think the design will only ever call for one so I can change this

Reply:
Fixed [here](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/159/commits/8a7e632d37f2f5232d0350324b01f07443480a44)